src/control/createIndex.ts

```diff
@@ -26,9 +28,9 @@ import {
 /**
  * @see [Understanding indexes](https://docs.pinecone.io/docs/indexes)
  */
-export interface CreateIndexOptions extends Omit<CreateIndexRequest, 'metric'> {
-  /** The distance metric to be used for similarity search. You can use 'euclidean', 'cosine', or 'dotproduct'. Defaults to 'cosine'. */
-  metric?: CreateIndexRequestMetricEnum;
+export interface CreateIndexOptions extends Omit<CreateIndexRequest, 'spec'> {
+  /** This option specifies how the index should be deployed. */
+  spec: CreateIndexSpec;
```

Comment:
Two things here:

`metric` used to be marked as required which is why we overwrote it to optional. This is no longer necessary. Instead we're overwriting `spec` with our custom types below specifically for creating indexes.